findspam.py

```diff
@@ -903,15 +903,15 @@ def pattern_email(s, site):
 def keyword_link(s, site):   # thanking keyword and a link in the same short answer
     if len(s) > 400:
         return False, ""
-    link = regex.compile(r'(?i)<a href="https?://\S+').search(s)
+    link = regex.compile(r'(?i)<a href="https?://[\w_-]++').search(s)
```

Comment:
The `_` character is part of the word, `\w`, character class. I believe the intent here was to get the entirety of the URL, which can include characters other than `[\w-]`, so this change doesn't do what was intended.